spec/jobs/import_url_job_spec.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ImportUrlJob do
+  let(:user) { create(:user) }
+
+  let(:file_path) { fixture_path + '/4-20.png' }
+  let(:file_hash) { '/673467823498723948237462429793840923582' }
+
+  let(:file_set) do
+    FileSet.new(import_url: "http://example.org#{file_hash}",
+                label: file_path) do |f|
+      f.apply_depositor_metadata(user.user_key)
+    end
+  end
+
+  let(:operation) { create(:operation) }
+  let(:success_service) { instance_double(Hyrax::ImportUrlSuccessService) }
+  let(:actor) { instance_double(Hyrax::Actors::FileSetActor, create_content: true) }
+
+  before do
+    allow(Hyrax::ImportUrlSuccessService).to receive(:new).and_return(success_service)
+    allow(Hyrax::Actors::FileSetActor).to receive(:new).with(file_set, user).and_return(actor)
+  end
+
+  context 'after running the job' do
+    before do
+      file_set.id = 'abc123'
+      allow(file_set).to receive(:reload)
+    end
+
+    it 'creates the content and updates the associated operation' do
+      WebMock.stub_request(:get, URI("http://example.org#{file_hash}")).to_return(
+        body: File.open(File.expand_path(file_path, __FILE__)).read, status: 200, headers: { 'Content-Type' => 'image/png', 'location' => URI("http://example.org") }
+      )
+
+      expect(success_service).to receive(:call)
+      expect(actor).to receive(:create_content).with(Tempfile, 'original_file', false).and_return(true)
+      described_class.perform_now(file_set, operation)
+      expect(operation).to be_success
+    end
+  end
+
+  context "when provider errors" do
+    before do
+      file_set.id = 'abc123'
+      allow(file_set).to receive(:reload)
+
+      WebMock.stub_request(:get, URI("http://example.org#{file_hash}")).to_return(
+        body: File.open(File.expand_path(file_path, __FILE__)).read, status: 500, headers: { 'Content-Type' => 'image/png', 'location' => URI("http://example.org") }
+      )
+    end
+
+    it 'does not return the url' do
+      expect(operation).not_to be_success
```

Comment:
Since this is job spec, I added a success/failure test instead of a status_code test.